nntrainer/layers/conv2d_layer.cpp

```diff
@@ -164,135 +209,47 @@ void Conv2DLayer::forwarding(bool training) {
 
     Tensor in_sub = input_.getBatchSlice(b, 1);
 
-    START_PROFILE(im2col_key);
-    im2col(in_sub, filter_dim, padding, stride, true, im2col_result);
-    END_PROFILE(im2col_key);
+    im2col(in_sub, filter_dim, padding, stride, {1, 1}, true, im2col_result);
 
-    START_PROFILE(conv_gemm_profile_key);
     filter_kernel.dot(im2col_result, out, false, true);
-    END_PROFILE(conv_gemm_profile_key);
   }
 
   filter_kernel.reshape(filter_dim);
-  START_PROFILE(add_bias_key);
   status = hidden_.add_i(bias_kernel);
   if (status != ML_ERROR_NONE) {
     throw std::invalid_argument("[Conv2D] adding bias failed");
   }
-  END_PROFILE(add_bias_key);
-
   loss = 0.0f;
   if (weight_regularizer == WeightRegularizerType::l2norm) {
     loss += weight_regularizer_constant * 0.5f * (filter_kernel.l2norm());
   }
 }
 
 void Conv2DLayer::calcDerivative() {
-
-  TensorDim &in_dim = input_dim[0];
-
   Tensor &derivative = net_hidden[0]->getGradientRef();
+  Tensor &input_derivative = net_input[0]->getGradientRef();
   Tensor &filter_kernel = weightAt(ConvParams::weight).getVariableRef();
 
-  std::array<unsigned int, CONV2D_DIM> same_pad;
-  same_pad[0] = kernel_size[0] - 1;
-  same_pad[1] = kernel_size[1] - 1;
-
-  /** Calculate return derivative
```

Comment:
is it intended to remove this description? or if the procedure has changed, update the description than deleting it.

Reply:
The procedure is deleted, so the description is removed

before) we calculate derivative out of conv operation
now) we calculated
col_matrix = w_trans (dot) derivative
return_derivative = col2im(col_matrix)

Now, the code itself became pretty much self-descriptive, I'll add a small comment here.